### PR TITLE
Add Permission Policy

### LINKED ISSUE #78: Do we need a permissions policy for third-party frames?
The digital identity draft - https://wicg.github.io/digital-identities/ - does not specify whether the digital identity API should be callable from an <iframe>

Should there be a digital-identity-specific permission policy - <iframe allow=""> which enables a toplevel frame to grant an <iframe> the ability to make a digital identity request?

### PR DESCRIPTION
**Pull request by marcoscaceres**
Closes #78 

Depends on #129 

The following tasks have been completed:

- [ ] Modified Web platform tests (link)

Implementation commitment:

- [ ] [WebKit](https://bugs.webkit.org/show_bug.cgi?id=275783)
- [ ] Chromium (link to issue)
- [ ] Gecko (link to issue)

Documentation and checks

- [ ] Affects privacy
- [ ] Affects security
- [ ] Pinged MDN
- [ ] Updated Explainer

***
<a href="https://pr-preview.s3.amazonaws.com/WICG/digital-credentials/pull/132.html" title="Last updated on Jul 5, 2024, 12:00 AM UTC (d4291be)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/digital-credentials/132/69a62f2...d4291be.html" title="Last updated on Jul 5, 2024, 12:00 AM UTC (d4291be)">Diff</a>

### CODE REVIEW

**Comment by marcoscaceres** [comment]
This obviously depends on https://github.com/WICG/digital-credentials/pull/129 .... will slot this into the algorithm after 129 in merged.

**Comment by marcoscaceres** [comment]
Webkit tracking bug https://bugs.webkit.org/show_bug.cgi?id=275783

File: index.html [hunk 1]
```diff
@@ -272,6 +272,16 @@ <h3>
       The <dfn data-dfn-for="DigitalCredential">data</dfn> member is the
       credential's response data.
     </p>
+    <h3>
+      [[\DiscoverFromExternalSource]](origin, options, sameOriginWithAncestors)
+      internal method
```
Comment (RByers) [inline index.html:277]: WebAuthn [seems to say](https://w3c.github.io/webauthn/#sctn-permissions-policy) we can't do this here due to lack of access to the settings object. I haven't looked into the details though, WDYT?

**Review by RByers — COMMENTED** [review]
Thanks! Looks good overall, just a question about integration here vs. how it's done for WebAuthn.

File: index.html (hunk 1 above)
Comment (marcoscaceres) [reply index.html:277]: Ah, right! I need to add it to the Cred Man itself! That's what they do for WebAuthn. I'll do that.

File: index.html [hunk 2]
```diff
@@ -309,6 +309,29 @@ <h3>
...
+    <section class="informative">
+      <h3>
+        User consent
+      </h3>
+      <p>
+        The <cite>Digital Credential API</cite> is a [=powerful feature=] that
+        requires [=express permission=] from an end-user. This requirement is
+        normatively enforced when calling `navigator.identity.get()` via the
+        {{DigitalCredential\[[DiscoverFromExternalSource]](origin, options,
+        sameOriginWithAncestors)}} internal method.
```
Comment (marcoscaceres) [inline index.html:321]:
```suggestion
        normatively enforced when calling {{CredentialsContainer}}'s {{CredentialsContainer/get()}}
        method.
```

File: index.html [hunk 3]
```diff
@@ -309,6 +309,28 @@ <h3>
...
     </p>
+    <section class="informative">
+      <h3>
+        User consent
+      </h3>
+      <p>
+        The <cite>Digital Credential API</cite> is a [=powerful feature=] that
+        requires [=express permission=] from an end-user. This requirement is
+        normatively enforced when calling {{CredentialsContainer}}'s {{CredentialsContainer/get()}}
+        method.
```
Comment (marcoscaceres) [inline index.html:320]:
```suggestion
        normatively enforced when calling {{CredentialsContainer}}'s
        {{CredentialsContainer/get()}} method.
```

File: index.html [hunk 4]
```diff
@@ -309,6 +309,28 @@ <h3>
...
+      </p>
+    </section>
+    <section id="permissions-policy" data-cite="permissions-policy">
+      <h2>
+        Permissions Policy integration
+      </h2>
+      <p>
+        This specification defines a [=policy-controlled feature=] identified
+        by the string <dfn class="permission">"digital-credentials-get"</dfn>
+        [[permissions-policy]]. Its [=policy-controlled feature/default
```
Comment (marcoscaceres) [inline index.html:330]:
```suggestion
        by the string <dfn class="permission">"digital-credentials-get"</dfn>. 
        Its [=policy-controlled feature/default
```

File: index.html [hunk 5]
```diff
@@ -309,6 +309,28 @@ <h3>
...
+    </section>
+    <section id="permissions-policy" data-cite="permissions-policy">
+      <h2>
+        Permissions Policy integration
+      </h2>
+      <p>
+        This specification defines a [=policy-controlled feature=] identified
+        by the string <dfn class="permission">"digital-credentials-get"</dfn>. 
+        Its [=policy-controlled feature/default
+        allowlist=] is [=allowlist/'self'=].
```
Comment (marcoscaceres) [inline index.html:331]:
```suggestion
        Its [=policy-controlled feature/default allowlist=] is [=allowlist/'self'=].
```

File: index.html (hunk 1 above)
Comment (marcoscaceres) [reply index.html]: @RByers sent https://github.com/w3c/webappsec-credential-management/pull/242

**Review by samuelgoto — APPROVED** [review]
LGTM

File: index.html [hunk 6]
```diff
@@ -309,6 +319,29 @@ <h3>
...
+        sameOriginWithAncestors)}} internal method.
+      </p>
+    </section>
+    <section id="permissions-policy" data-cite="permissions-policy">
+      <h2>
+        Permissions Policy integration
+      </h2>
+      <p>
+        This specification defines a [=policy-controlled feature=] identified
+        by the string <dfn class="permission">"digital-credentials-get"</dfn>
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply index.html]: Great. I'll add WebKit there too.

**Comment by marcoscaceres** [comment]
Ok, cool. I'll add some tests.

**Comment by marcoscaceres** [comment]
I'll send the test up from the WebKit patch. They should land tomorrow, but I'll go ahead and merge this now.

**Comment by marcoscaceres** [comment]
Tests for permission policy can be found here: https://github.com/web-platform-tests/wpt/pull/47662/

**Pull request merged**